### PR TITLE
feat: react-data-grid-react-window should be complete extension package

### LINKED ISSUE #25461: DataGrid: Implement or document a virtualized DataGridBody
The prototype in https://github.com/microsoft/fluentui/pull/25238 has an example of a virtualized `DataGridBody` powered by recomposition and `react-virtual`.

Either implement a variant of `DataGridBody` that is virtualized or add documentation on how to do that

### PR DESCRIPTION
**Pull request by ling1726**
Implements `aria-rowcount` and `aria-rowindex` support for the virtualized DataGrid.

This PR also turns the package into a 'complete' extension package so that all DataGrid components are re-exported from it whether it is recomposed or not. This package should never be exported from react-components suite.

Fixes #25461

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/f87e6c14d226f9a73bbf9a89fe9858532cd47ace'>f87e6c14d226f9a73bbf9a89fe9858532cd47ace</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/Avatar-mount_base.html">1249</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/Avatar-mount.html">1313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/Button-mount_base.html">943</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/Button-mount.html">940</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProvider-mount_base.html">1542</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProvider-mount.html">1654</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProviderWithTheme-mount_base.html">638</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProviderWithTheme-mount.html">558</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">591</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">631</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">620</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/InfoButton-mount_base.html">567</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/InfoButton-mount.html">580</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/MakeStyles-mount_base.html">1923</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/MakeStyles-mount.html">1912</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/Persona-mount_base.html">3064</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/Persona-mount.html">3008</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/SpinButton-mount_base.html">2499</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26627/perf-test-react-components/SpinButton-mount.html">2567</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-table/src/components/DataGrid/useDataGridContextValues.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { useTableContextValues_unstable } from '../Table/useTableContextValues';
 import { DataGridContextValues, DataGridState } from './DataGrid.types';
 
+/**
+ * @internal
+ */
```
Comment (ling1726) [inline packages/react-components/react-table/src/components/DataGrid/useDataGridContextValues.ts:6]:
```suggestion
```

File: scripts/storybook/src/utils.js [hunk 2]
```diff
@@ -170,6 +170,9 @@ function _createCodesandboxRule(allPackageInfo = getAllPackageInfo()) {
 
       return acc;
     }, /** @type import('storybook-addon-export-to-codesandbox').BabelPluginOptions*/ ({}));
+
+    delete importMappings['@fluentui/react-data-grid-react-window'];
```
Comment (ling1726) [inline scripts/storybook/src/utils.js:174]: This is kind of a hack, but AFAIK there's no other way to distinguish a converged package that won't be in react-components vs a converged pacakge that will be in react-components

any suggestions @Hotell ?

File: packages/react-components/react-data-grid-react-window/src/components/DataGrid/DataGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  renderDataGrid_unstable,
+  useDataGridStyles_unstable,
+  useDataGridContextValues_unstable,
+} from '@fluentui/react-table';
+import type { DataGridProps } from '@fluentui/react-table';
+import type { ForwardRefComponent } from '@fluentui/react-utilities';
+
+/**
+ * DataGrid component - TODO: add more docs
+ */
```
Comment (layershifter) [inline packages/react-components/react-data-grid-react-window/src/components/DataGrid/DataGrid.tsx:13]:
```suggestion
```

**Review by layershifter — APPROVED** [review]

File: scripts/storybook/src/utils.js [hunk 4]
```diff
@@ -170,6 +170,13 @@ function _createCodesandboxRule(allPackageInfo = getAllPackageInfo()) {
 
       return acc;
     }, /** @type import('storybook-addon-export-to-codesandbox').BabelPluginOptions*/ ({}));
+
+    return {
+      ...importMappings,
+      '@fluentui/react-data-grid-react-window': {
```
Comment (Hotell) [inline scripts/storybook/src/utils.js:176]: we should come up with something more robust, as hardcoding this mapping for every extension package will not scale

**Review by Hotell — APPROVED** [review]

File: scripts/storybook/src/utils.js (hunk 4 above)
Comment (ling1726) [reply scripts/storybook/src/utils.js:176]: Tracking in #26691

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Pull request merged**